### ISSUE TITLE
Active quote is not updated when the customer changes account information

### ISSUE THREAD

**Issue by dyuk1987** [magento/magento2#21502]
### Summary (*)

I am not sure if the following issue should be regarded as a "bug", so I decided to put it here.
**Steps to reproduce the issue:**
_(Tested on a clean Magento 2.2.6 instance)_
1. Customer logs into his/her account, say john.doe@example.com
2. Customer adds a few products to the shopping cart
3. Customer changes the email address, say john.doe+2@example.com
4. Customer continues with the checkout process and successfully places an order
5. Order email is being sent to john.doe@example.com instead of john.doe+2@example.com

### Examples (*)

Just follow the steps above. 
The same applies to first name and last name as well.

### Proposed solution

This issue should be due to the fact that the email address, first name as well as last name are being saved to the quote once the cart is created, and further modifications on these values are not reflected and updated to the quote.

I have created an observer to solve this issue for the time being.

**events.xml**
```
<event name="customer_save_after_data_object">
    <observer name="sync_quote_customer_details" instance="Namespace\Module\Observer\SyncQuoteCustomerDetails" />
</event>
```

**SyncQuoteCustomerDetails.php**
```
class SyncQuoteCustomerDetails implements ObserverInterface
{
    protected $quoteRepository;

    public function __construct(
        \Magento\Quote\Api\CartRepositoryInterface $quoteRepository
    ) {
        $this->quoteRepository = $quoteRepository;
    }

    public function execute(Observer $observer) {
        $event = $observer->getEvent();
        $customerDataObj = $event->getData('customer_data_object');
				
        try {
            $quote = $this->quoteRepository->getForCustomer($customerDataObj->getId());
						
            $quote->setCustomerEmail($customerDataObj->getEmail());
            $quote->setCustomerFirstname($customerDataObj->getFirstname());
            $quote->setCustomerLastname($customerDataObj->getLastname());
						
            $this->quoteRepository->save($quote);
        } catch (\Exception $e) {
            //do nothing if no active quote is found
        }
    }
}
```

**Comment by magento-engcom-team** [comment]
Hi @dyuk1987. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@dyuk1987 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
HI @dyuk1987 thank you for you report, this issue has already fixed in 2.3-develop branch. and will be available on 2.3.1 release
![deepinscreenshot_select-area_20190228143934](https://user-images.githubusercontent.com/40756439/53567046-ae63f480-3b66-11e9-9587-bf49eaa05132.png)
